src/compute/api.rs

```diff
@@ -53,151 +56,253 @@ pub fn create_keypair(session: &Session, request: KeyPairCreate) -> Result<KeyPa
 
     debug!("Creating a key pair with {:?}", request);
     let body = KeyPairCreateRoot { keypair: request };
-    let root: KeyPairRoot = session.post_json(COMPUTE, &["os-keypairs"], body, version)?;
+    let mut builder = session.post(COMPUTE, &["os-keypairs"]).json(&body);
+
+    if let Some(version) = version {
+        builder = builder.api_version(version)
+    }
+
+    let root: KeyPairRoot = builder.fetch().await?;
     debug!("Created key pair {:?}", root.keypair);
     Ok(root.keypair)
 }
 
 /// Create a server.
-pub fn create_server(session: &Session, request: ServerCreate) -> Result<Ref> {
+pub async fn create_server(session: &Session, request: ServerCreate) -> Result<Ref> {
     debug!("Creating a server with {:?}", request);
     let body = ServerCreateRoot { server: request };
-    let root: CreatedServerRoot = session.post_json(COMPUTE, &["servers"], body, None)?;
+    let root: CreatedServerRoot = session
+        .post(COMPUTE, &["servers"])
+        .json(&body)
+        .fetch()
+        .await?;
     trace!("Requested creation of server {:?}", root.server);
     Ok(root.server)
 }
 
 /// Delete a key pair.
-pub fn delete_keypair<S: AsRef<str>>(session: &Session, name: S) -> Result<()> {
+pub async fn delete_keypair<S: AsRef<str>>(session: &Session, name: S) -> Result<()> {
     debug!("Deleting key pair {}", name.as_ref());
-    let _ = session.delete(COMPUTE, &["os-keypairs", name.as_ref()], None)?;
+    let _ = session
+        .delete(COMPUTE, &["os-keypairs", name.as_ref()])
+        .fetch()
+        .await?;
     debug!("Key pair {} was deleted", name.as_ref());
     Ok(())
 }
 
 /// Delete a server.
-pub fn delete_server<S: AsRef<str>>(session: &Session, id: S) -> Result<()> {
+pub async fn delete_server<S: AsRef<str>>(session: &Session, id: S) -> Result<()> {
     trace!("Deleting server {}", id.as_ref());
-    let _ = session.delete(COMPUTE, &["servers", id.as_ref()], None)?;
+    let _ = session
+        .delete(COMPUTE, &["servers", id.as_ref()])
+        .fetch()
```

Comment:
I think the CI job fails because of this. fetch() expects a JSON, deletion requests normally don't provide one. You need to use `send` instead of `fetch` (same everywhere else).

As an aside, I wonder why it even compiles. How does it derive the type of `_` in this case?